explorer/parse_and_execute/parse_and_execute.cpp

```diff
@@ -0,0 +1,96 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "explorer/parse_and_execute/parse_and_execute.h"
+
+#include <locale>
+
+#include "common/check.h"
+#include "common/error.h"
+#include "explorer/interpreter/exec_program.h"
+#include "explorer/interpreter/stack_space.h"
+#include "explorer/syntax/parse.h"
+#include "explorer/syntax/prelude.h"
+#include "llvm/ADT/ScopeExit.h"
+
+namespace Carbon {
+
+// Returns a scope exit function for printing the timing of a step on scope
+// exit. Note the use prints step timings in reverse order.
+static auto PrintTimingOnExit(TraceStream* trace_stream, const char* label,
```

Comment:
Here again, this seems pretty unrelated to the topic of the PR. Did you intend to include it, or did a merge go awry or something? If it's intentional, it should probably be mentioned in the PR description.